client/src/providers/UserProvider.tsx

```diff
@@ -42,10 +47,14 @@ export function UserProvider(props: React.PropsWithChildren<{}>): JSX.Element {
   ): Promise<void> {
     let newUser: User | null = user;
     if (shouldFetch) {
+      setIsLoading(true);
       newUser = await fetchUser();
     }
+
     setUser(newUser);
 
+    if (isLoading) setIsLoading(false);
```

Comment:
so react states actually work async under the hood, the do some batching and stuff and basically we can't rely on the `isLoading` state to be `true` here
but i do think we could just set it to false without the condition as regardless of whether we even fetched, it should still be false at this point